syntax = "proto3";

service DataNodeService {
  //store a block of data
  rpc WriteBlock (WriteBlockRequest) returns (WriteBlockResponse);
  
  //read a block of data
  rpc ReadBlock (ReadBlockRequest) returns (ReadBlockResponse);
}

message WriteBlockRequest {
  string block_id = 1;   //identifier of the block
  bytes data = 2;        //binary data of the block
}

message WriteBlockResponse {
  string status = 1;     //status of the block storage (success or error)
}

message ReadBlockRequest {
  string block_id = 1;   //identifier of the block
}

message ReadBlockResponse {
  bytes data = 1;        //binary data of the requested block
  string status = 2;     //status of the operation (success or error)
}
